chapter-06/inprogress/bookworm_internal_test.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"reflect"
+	"testing"
+)
+
+func TestLoadBookworms(t *testing.T) {
+	noError := func(err error) bool { return err == nil }
+
+	tests := map[string]struct {
+		bookwormsFile string
+		want          []Bookworm
+		checkError    func(err error) bool
+	}{
+		"no common book": {
+			bookwormsFile: "testdata/no_common_book.json",
+			want:          bookwormsWithNoCommonBooks,
+			checkError:    noError,
+		},
+		"file doesn't exist": {
+			bookwormsFile: "testdata/no_file_here.json",
+			checkError: func(err error) bool {
+				return errors.Is(err, fs.ErrNotExist)
+			},
+		},
+		"invalid JSON": {
+			bookwormsFile: "testdata/invalid.json",
+			checkError: func(err error) bool {
+				var expectedErr *json.SyntaxError
+				return errors.As(err, &expectedErr)
```

Comment:
Using `.As` isn't really making use of As if we're not using the fields of the expectedErr. Instead, we should be using `Is` here.

If we want to use `As`, we can check the `expectedErr.Offset` field.

Reply:
We just wanted to check we are expecting the right type of error, are you saying we should be more precise with the offset?

Reply:
I don't think it completely makes sense to check that we received the right _type_. I think we should check that we received a specific _ErrJSON..._, or that we received a SyntaxError at a specific offset.